secure/key-manager/src/tests.rs

```diff
@@ -37,13 +37,32 @@ struct Node {
 }
 
 fn setup_secure_storage(config: &NodeConfig) -> InMemoryStorage {
+    let mut sec_storage = InMemoryStorage::new();
     let test_config = config.clone().test.unwrap();
-    let mut keypair = test_config.consensus_keypair.unwrap();
-    let prikey = Value::Ed25519PrivateKey(keypair.take_private().unwrap());
 
-    let mut sec_storage = InMemoryStorage::new();
+    let mut a_keypair = test_config.account_keypair.unwrap();
+    let a_prikey = Value::Ed25519PrivateKey(a_keypair.take_private().unwrap());
+
+    sec_storage
+        .create(crate::ACCOUNT_KEY, a_prikey, &Policy::public())
+        .unwrap();
+
+    let mut c_keypair = test_config.consensus_keypair.unwrap();
+    let c_prikey = c_keypair.take_private().unwrap();
+    let c_prikey0 = Value::Ed25519PrivateKey(c_prikey.clone());
+    let c_prikey1 = Value::Ed25519PrivateKey(c_prikey);
+
+    sec_storage
+        .create(crate::CONSENSUS_KEY, c_prikey0, &Policy::public())
+        .unwrap();
+    // Ugly hack but we need this until we support retrieving a policy from within storage and that
+    // currently is not easy, since we would need to convert from Vault -> Libra policy.
     sec_storage
-        .create(crate::CONSENSUS_KEY, prikey, &Policy::public())
+        .create(
+            &format!("{}_previous", crate::CONSENSUS_KEY),
```

Comment:
which part is ugly hack? filling _previous? Policy::public?

Reply:
Yes, ideally this should be handled behind the scenes when generating the initial public key... creating the proper versioning at that point in time... or at least have a default.